src/main/java/net/sf/mpxj/ActivityCode.java

```diff
@@ -132,10 +146,17 @@ public List<ActivityCodeValue> getValues()
       return m_values;
    }
 
-   private final Integer m_uniqueID;
-   private final ActivityCodeScope m_scope;
-   private final Integer m_scopeUniqueID;
-   private final Integer m_sequenceNumber;
-   private final String m_name;
+   @Override public int compareTo(ActivityCode o)
+   {
+      int id1 = NumberHelper.getInt(getUniqueID());
+      int id2 = NumberHelper.getInt(o.getUniqueID());
+      return (Integer.compare(id1, id2));
+   }
+
+   private Integer m_uniqueID;
```

Comment:
General question: the PR switches a couple of classes from being immutable to mutable. I was just wondering if there was a specific reason for this? I appreciate that the MPXJ codebase is not consistent in this respect (20 years of legacy code!) but it's something I think about when I'm adding new classes. Interested to hear you thoughts.

Reply:
I prefer having data classes be immutable, we changed this to mimic the same pattern that was being used for Task and Resources. We'd be happy to update this to be immutable if that is the path we want to go down. I know for Tasks that would be difficult because of the number of parameters. Typically when I have a large number of parameters I like to employ a builder pattern. If that's something you are open to, we can do that here and then over time get the others to mimic the same pattern.